### ISSUE TITLE
css:S4664 rule causing violations in LESS files for nested modifiers

### ISSUE THREAD

**Issue by thomasoreilly** [SonarSource/sonar-css#156]
Hey, 

We currently using this rule but it's throwing a violation in our LESS files that doesn't seem right. 
The following code is invalid. 
SonarQube is saying:
"Expected selector ".my-class" to come before selector ".my-class:hover"
"Expected selector ".my-class" to come before selector ".my-class:active"
"Expected selector ".my-class" to come before selector ".my-class:focus"

Can anybody expand why this is? 

.my-class {
   ...
    &:hover, &:active, &:focus {
        ...
    }
}

Thanks

**Comment by vilchik-elena** [comment]
I didn't manage to reproduce with your code, but I did with something like this 
```less
.a {
   &:foo {}
}

.a { // Expected selector ".a" to come before selector ".a:foo" (no-descending-specificity)
}
```

Is it your case?

**Comment by thomasoreilly** [comment]
Hey @vilchik-elena. 

Apologies for the late response. 

We were getting issues similar to what you have there. 
It doesn't seem to play well with the less parent keyword, &.

We ended up just excluding our files from being scanned with these violations. 

Is this rule fundamentally broken with less?

**Comment by vilchik-elena** [comment]
@thomasoreilly In my example having that issue seems valid to me. Could you give complete example of the case you consider false positive and explain why it is false positive to you?
